Src/VimCore/ModeLineInterpreter.fs

```diff
@@ -0,0 +1,186 @@
+                true
+            elif not (Regex.Match(settingName, _settingName).Success) then
+
+                // Don't ignore illegal setting names.
+                false
+            elif Option.isNone localSetting then
+
+                // Ignore unsupported settings.
+                true
+            else
+                false
+
+        // Whether we should allow the setting
+        let shouldAllowSetting settingName =
+
+            // For security reasons, we disallow certain local settings.
+            let globalSetting = _globalSettings.GetSetting settingName
+            let localSetting = _localSettings.GetSetting settingName
+            if Option.isSome globalSetting then
+
+                // Disallow all global settings.
+                false
+            elif Option.isSome localSetting then
+
+                // Allow any local setting that isn't insecure.
+                _insecureLocalSettingNames
+                |> Seq.contains localSetting.Value.Name
+                |> not
+            else
+                false
+
+        // Process a single option like 'ts=8'.
+        let processOption (option: string) =
+            let option = option.Trim()
+
+            // Determine what kind of option this is.
+            let settingName, setter =
+                let m = Regex.Match(option, _assignment)
+                if m.Success then
+
+                    // The option is an assigned setting.
+                    let settingName = m.Groups.[1].Value
+                    let strValue = m.Groups.[2].Value
+                    let setter = fun () -> _localSettings.TrySetValueFromString settingName strValue
+                    settingName, setter
+                elif option.StartsWith("no") then
+
+                    // The option toggles the setting off.
+                    let settingName = option.Substring(2)
+                    let toggleOff = (SettingValue.Toggle false)
+                    let setter = fun () ->_localSettings.TrySetValue settingName toggleOff
+                    settingName, setter
+                else
+
+                    // The option toggles the setting on.
+                    let settingName = option
+                    let toggleOn = (SettingValue.Toggle true)
+                    let setter = fun () -> _localSettings.TrySetValue settingName toggleOn
+                    settingName, setter
+
+            // Check whether we should apply the setter.
+            if shouldIgnoreSetting settingName then
+                true
+            elif shouldAllowSetting settingName then
+                setter()
+            else
+                false
+
+        // Split the options string into fields.
+        let splitFields (options: string) =
+            options.Replace(@"\:", ":").Split(' ', '\t')
+
+        // Process the "first" format of modeline, i.e. "vim: ...".
+        let processFirstForm modeLine =
+            let m = Regex.Match(modeLine, _firstPattern)

```

Comment:
This means that for every line we're constructing the `Regex` object for `_firstPattern` and then executing it fresh against the current line. Worried that could be expensive. Did you consider making the fields be of type `Regex` and using the compiled option to ensure speedy execution here?

Reply:
That's a good point; I'll make that change.